main.go

```diff
@@ -0,0 +1,226 @@
+			return // Exit if client initialization fails
+		}
+		defer client.Close()
+
+		contentGenerator = &GeminiLLM{
+			Client: client,
+			Model:  *model,
+		}
+	default:
+		logger.Error("Invalid language model provider", "provider", *llmProvider)
+		return
+	}
+
+	if *queryFromCmd != "" {
+		query := *queryFromCmd
+		agent := Agent{
+			Model:            *model,
+			Query:            query,
+			ContentGenerator: contentGenerator,
+			MaxIterations:    *maxIterations,
+			tracePath:        *tracePath,
+			promptFilePath:   *promptFilePath,
+			Kubeconfig:       *kubeconfig,
+			RemoveWorkDir:    *removeWorkDir,
+			templateFile:     *templateFile,
+			markdownRenderer: mdRenderer,
+		}
+		agent.Execute(ctx)
+		return
+	}
+
+	chatSession := session{
+		Queries: []string{},
+		Model:   *model,
+	}
+
+	fmt.Printf("\033[31mHey there, what can I help you with today?\033[0m")
+	for {
+		fmt.Printf("\n>> ")
+		reader := bufio.NewReader(os.Stdin)
+		query, err := reader.ReadString('\n')
+		if err != nil {
+			fmt.Println("Error reading input:", err)
+			return
+		}
+		query = strings.TrimSpace(query)
+
+		if query == "" {
+			continue
+		}
+		switch query {
+		case "reset":
+			chatSession.Queries = []string{}
+			clearScreen()
+		case "clear":
+			clearScreen()
+		case "exit", "quit":
+			fmt.Println("Allright...bye.")
+			return
+		case "models":
+			modelNames, err := contentGenerator.ListModels(ctx)
+			if err != nil {
+				fmt.Println("Error listing models:", err)
+				continue
+			}
+			fmt.Println("Available models:")
+			for _, modelName := range modelNames {
+				fmt.Println(modelName)
+			}
+		default:
+			if strings.HasPrefix(query, "model") {
+				parts := strings.Split(query, " ")
+				if len(parts) > 2 {
+					fmt.Println("Invalid model command. expected format: model <model-name>")
+					continue
+				}
+				if len(parts) == 1 {
+					out := fmt.Sprintf("Current model is `%s`\n", chatSession.Model)
+					rendered, _ := mdRenderer.Render(out)
+					fmt.Println(rendered)
+					continue
+				}
+				chatSession.Model = parts[1]
+				fmt.Printf("Model set to `%s`\n", chatSession.Model)
+				continue
+			}
+			agent := Agent{
+				Model:            chatSession.Model,
+				Query:            query,
+				PastQueries:      chatSession.PreviousQueries(),
+				ContentGenerator: contentGenerator,
+				MaxIterations:    *maxIterations,
+				tracePath:        *tracePath,
+				promptFilePath:   *promptFilePath,
+				Kubeconfig:       *kubeconfig,
+				RemoveWorkDir:    *removeWorkDir,
+				templateFile:     *templateFile,
+				markdownRenderer: mdRenderer,
+			}
+			agent.Execute(ctx)
+			chatSession.Queries = append(chatSession.Queries, query)
+		}
+	}
+}
+
+// session represents each the chat session.
+type session struct {
+	Queries []string `json:"queries"`
+	Model   string   `json:"model"`
+}
+
+func (s *session) PreviousQueries() string {
+	return strings.Join(s.Queries, "\n")
+}
+
+// Logger key for context
+type loggerKey struct{}
```

Comment:
Cool.  We may be able to use `klog.FromContext`, but I think we should merge this PR and then move this into its own package.